main.ts

```diff
@@ -99,25 +99,28 @@ function removeCache(messages_only_content: string[]) {
 
 async function fetchFull(chat: Chat, messages: Messages) {
   let message: Response | undefined;
-  let text: string | undefined;
+  let text: string = "";
   let messageData: MessageData | undefined;
 
   for (let i = 0; i < messages.length; i += 2) {
-    text = "";
```

Comment:
The `text` must be reset at each loop so that the text is not stacked on top of each other.